optaplanner-core/src/main/java/org/optaplanner/core/api/score/buildin/hardsoftdouble/HardSoftDoubleScoreHolder.java

```diff
@@ -25,14 +25,18 @@
 import org.optaplanner.core.api.domain.constraintweight.ConstraintConfiguration;
 import org.optaplanner.core.api.domain.constraintweight.ConstraintWeight;
 import org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScoreHolder;
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScoreHolder;
 import org.optaplanner.core.api.score.holder.AbstractScoreHolder;
 
 /**
  * WARNING: NOT RECOMMENDED TO USE DUE TO ROUNDING ERRORS THAT CAUSE SCORE CORRUPTION.
  * Use {@link HardSoftBigDecimalScoreHolder} instead.
  *
  * @see HardSoftDoubleScore
+ * @deprecated Double-based scores are deprecated as floating point numbers can't represent a decimal number correctly.
+ *             Use {@link HardSoftBigDecimalScoreHolder} or a scaled {@link HardSoftLongScoreHolder} instead.
  */
+@Deprecated(/* forRemoval = true */)
```

Comment:
I don't mind the "forRemoval" comment, but note that we intend to remove all existing Deprecated methods in 8.0 and those don't have that forRemoval comment.